script/publish_public_data_json

```diff
@@ -0,0 +1,191 @@
+    """Converts ndjson files on GCP to json files."""
+    storage_client = storage.Client()
+
+    blobs = storage_client.list_blobs(bucket, prefix=gcp_path)
+
+    for blob in blobs:
+        content = blob.download_as_string().decode("utf-8").strip()
+        json_list = [json.loads(line) for line in content.split("\n")]
+
+        blob.upload_from_string(json.dumps(json_list, indent=2))
+
+
+def write_results_to_temp_table(client, project, query_file, date):
+    """
+    Write the results of the query to a temporary table and return the table
+    name.
+    """
+    (dataset, table, version) = dataset_table_version_from_file(query_file)
+
+    table_date = date.replace("-", "")
+    temp_table = f"{project}.{dataset}.{table}_{version}_{table_date}_temp"
+
+    with open(query_file) as query_stream:
+        sql = query_stream.read()
+        job_config = bigquery.QueryJobConfig(destination=temp_table)
+        query_job = client.query(sql, job_config=job_config)
+        query_job.result()
+
+        return temp_table
+
+
+def dataset_table_version_from_file(file_name):
+    """Extract the dataset, table and version from the provided file name."""
+
+    query_file_re = re.search(QUERY_FILE_RE, file_name)
+    if query_file_re:
+        dataset = query_file_re.group(1)
+        table = query_file_re.group(2)
+        version = query_file_re.group(3)
+
+        return (dataset, table, version)
+    else:
+        print("Invalid file naming format: {}", file_name)
+        return None
+
+
+def main():
+    args, query_arguments = parser.parse_known_args()
+
+    try:
+        metadata = Metadata.of_sql_file(args.query_file)
+    except FileNotFoundError:
+        print("No metadata file for: {}".format(args.query_file))
+        return
+
+    # check if the data should be published as JSON
+    if not metadata.is_public_json():
+        return
+
+    client = bigquery.Client(args.project_id)
+
+    if args.parameter:
+        date_search = re.search(SUBMISSION_DATE_RE, args.parameter)
+
+        if date_search:
+            date = date_search.group(1)
+
+    (dataset, table, version) = dataset_table_version_from_file(args.query_file)
+
+    if metadata.is_incremental():
+        if date is None:
+            print("Cannot publish JSON. submission_date needs to be set as parameter.")
+            sys.exit(1)
+
+        # if it is an incremental query, then the query result needs to be
+        # written to a temporary table to get exported as JSON
+        temp_result_table = write_results_to_temp_table(
+            client, args.project_id, args.query_file, date
+        )
+        publish_table_as_json(
+            args.target_bucket,
+            client,
+            args.api_version,
+            dataset,
+            table,
+            version,
+            temp_result_table,
+            date,
+        )
+
+        # remove the temporary table after it has been exported
+        client.delete_table(temp_result_table)
```

Comment:
If `publich_table_as_json` fails, we'll end up the tmp table still available in bq. One alternative could be to wrap the temp table pieces into an object with `__enter__` for setup and `__exit__` for teardown, using `with` here.